### DISCUSSION TITLE
How do you add keymaps for both format range + format whole document?

### DISCUSSION THREAD

**Discussion by idelice** [LazyVim/LazyVim#1662] [General]
RIght now, i can do space + c + f to format range, but not the whole document.

**Comment by dpetka2001** [comment]
What do you mean? `<C-s>` for saving the changes in the buffer already autoformats the whole buffer.

**Reply by idelice** [reply]
> What do you mean? `<C-s>` for saving the changes in the buffer already autoformats the whole buffer.

I set autoformat to false bc i want to manually format the document or the marked code.

**Reply by idelice** [reply]
> What do you mean? `<C-s>` for saving the changes in the buffer already autoformats the whole buffer.

In other words; i don't have any mappings to format document. I only have keymapping to format range

**Reply by folke** [reply]
`<leader>cf` also formats the whole document. It only formats a range in visual mode

**Reply by idelice** [reply]
> `<leader>cf` also formats the whole document. It only formats a range in visual mode

I haven't had luck formatting the document. Only the visual range. 

I may have to try again in a different environment and see what happens.

**Comment by amoschristian** [comment]
Up to this thread, since i experience similar but the other way around

WIth prettier + conform i can format whole document with `<leader>cf` but i cant do range formatting
whenever i pick some block of code using visual mode and format it, it will format whole document
